### PR TITLE
feat: Add common email domains validation on organization sharing

### PR DESCRIPTION
**Pull request by jnfrati**
Closes: #4770 

### Description:
Prevent users to set common email domains (e.g. gmail.com) as allowed email domains.

This PR covers:
1- Validation on PATCH /org/allowedEmailDomains
2- Validation on AllowedDomainsPanel (Org settings)
3- Removal of checkbox on UserCompletionModal in case the users email is from the common email domains list.

image
---

image
image
---

image
image

### CODE REVIEW

**Comment by jnfrati** [comment]
@ZeRego @hamzahc1 Here is the PR for the email validation issue! Let me know if there's anything I should fix for this one 👍

**Comment by jnfrati** [comment]
Sorry about the amount of commits on fixing the prettier formatting 🙇 It should be ready now 👍

File: packages/backend/src/services/OrganizationService/OrganizationService.ts [hunk 1]
```diff
@@ -325,6 +327,10 @@ export class OrganizationService {
             throw new ForbiddenError();
         }
 
+        if (data.emailDomains.some((domain) => isEmailProviderDomain(domain))) {
+            throw new ParameterError();
```
Comment (ZeRego) [inline packages/backend/src/services/OrganizationService/OrganizationService.ts:331]: We should also add this check in `packages/backend/src/services/UserService.ts` -> `completeUserSetup()`

**Review by ZeRego — DISMISSED** [review]
code looks good 🥇  just missing the parameter check in user service.

File: packages/backend/src/services/OrganizationService/OrganizationService.ts (hunk 1 above)
Comment (jnfrati) [reply packages/backend/src/services/OrganizationService/OrganizationService.ts:331]: Sounds good! Although here I'm not sure what kind of error it should be. Maybe `ParametersError` with a description like `email domain is not valid to enable domain organization access`?

Comment (ZeRego) [reply packages/backend/src/services/OrganizationService/OrganizationService.ts:331]: Yes, that would work 👍 We can add a similar message in the organization service error.

**Comment by jnfrati** [comment]
@ZeRego Changes ready! 
Let me know if I missed something, otherwise, should be ready to merge 👏

File: packages/frontend/src/utils/fieldValidators.ts [hunk 2]
```diff
@@ -36,13 +40,38 @@ export const isValidEmail: FieldValidator<string> = (fieldName) => (value) =>
...
+            const readableDomainList = new Intl.ListFormat('en-US', {
+                style: 'long',
+                type: 'conjunction',
+            }).format(invalidDomains);
+
+            return `
+                ${readableDomainList} ${
+                invalidDomains.length === 1 ? 'is' : 'are'
+            } not allowed as organization email
+            `;
```
Comment (IrakliJani) [inline packages/frontend/src/utils/fieldValidators.ts:74]: this chunk exists both in the frontend and the backend.
can you extract both and move it to `packages/commmon/src/utils/email.ts`?

maybe the whole validation not only this chunk.

**Review by IrakliJani — CHANGES_REQUESTED** [review]
great work! made one comment. lmk if you have questions

File: packages/frontend/src/utils/fieldValidators.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/utils/fieldValidators.ts]: looks great, im gonna run tests locally and merge 🥇 🥇 🥇 
thanks so much!

**Pull request merged**